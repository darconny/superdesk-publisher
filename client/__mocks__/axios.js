// import axios from "axios"

// // Mock out all top level functions, such as get, put, delete and post:
// jest.mock("axios")

// axios.get.mockImplementation(url => {
//     if (url === 'www.example.com') {
//         return Promise.resolve({ data: {...} })
//     } else {

//     }
// })

"use strict";
module.exports = {
  get: url => {
    let data = {};

    if (url.includes("tenants")) {
      data = {
        _embedded: {
          _items: [
            {
              id: 1,
              subdomain: "tenant1",
              domain_name: "sourcefabric.org",
              name: "TENANT1",
              code: "tenant1",
              organization: {
                id: 1,
                name: "test",
                code: "0gjdk8"
              },
              created_at: "2017-06-30T09:36:00+00:00",
              updated_at: "2019-04-16T19:45:19+00:00",
              enabled: true,
              theme_name: "swp/daily-theme",
              amp_enabled: true,
              output_channel: null,
              articles_count: 140,
              fbia_enabled: false,
              paywall_enabled: false
            },
            {
              id: 2,
              subdomain: "tenant2",
              domain_name: "sourcefabric.org",
              name: "TENANT2",
              code: "tenant2",
              organization: {
                id: 1,
                name: "test",
                code: "0gjdk8"
              },
              created_at: "2017-06-30T09:36:00+00:00",
              updated_at: "2019-04-16T19:45:19+00:00",
              enabled: true,
              theme_name: "swp/daily-theme",
              amp_enabled: true,
              output_channel: null,
              articles_count: 140,
              fbia_enabled: false,
              paywall_enabled: false
            }
          ]
        }
      };
    }

    if (url.includes("content/routes")) {
      data = {
        _embedded: {
          _items: [
            {
              id: 1,
              name: "route1"
            },
            {
              id: 2,
              name: "route2"
            }
          ]
        }
      };
    }

    if (url.includes("content/lists")) {
      data = {
        _embedded: {
          _items: [
            {
              id: 1,
              name: "content list 1",
              type: "manual"
            },
            {
              id: 2,
              name: "content list 2",
              type: "manual"
            }
          ]
        }
      };
    }

    if (url.includes("packages/seo")) {
      data = {
        meta_title: null,
        meta_description: null,
        og_title: null,
        og_description: null,
        twitter_title: null,
        twitter_description: null
      };
    }

    // console.log("\x1b[44m\x1b[37m%s\x1b[0m", "API URL: " + url);
    return Promise.resolve({ data: data });
  },

  post: url => {
    let data = {};

    if (url.includes("organization/articles/related")) {
      data = {
        related_article_items: [
          { tenants: [], title: "test related 1" },
          {
            tenants: [
              {
                code: "eif0ca"
              }
            ],
            title: "test related 2"
          }
        ]
      };
    }

    if (url.includes("auth/superdesk")) {
      data = {
        token: { api_key: "asdfasdf" }
      };
    }

    if (url.includes("preview/package/generate_token")) {
      data = {
        preview_url: "https://sourcefabric.org/preview"
      };
    }

    if (url.includes("organization/destinations")) {
      data = {};
    }

    // console.log("\x1b[44m\x1b[37m%s\x1b[0m", "API URL: " + url);
    return Promise.resolve({ data: data });
  }
};
