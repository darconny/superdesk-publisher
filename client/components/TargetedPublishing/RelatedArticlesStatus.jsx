import React from 'react';
import axios from 'axios';
import classNames from 'classnames';
import {ToggleBox} from '../UI/ToggleBox.jsx';
import Loading from '../UI/Loading/Loading.jsx';

export default class RelatedArticlesStatus extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            relatedArticles: [],
            loading: true
        };
    }

    componentDidMount() {
        if (!this.props.item.associations) return;
        this.getRelatedArticles();
    }

    componentWillReceiveProps({rules}) {
        this.setState({rules});
    }

    getRelatedArticles = () => {
        return axios.post(this.props.apiUrl + 'organization/articles/related/', this.props.item, {headers: this.props.apiHeader})
            .then(res => {
                this.setState({
                    relatedArticles: res.data.related_article_items,
                    loading: false
                });
                return res;
            });
    }

    render() {
        if (!this.props.item.associations) {
            return null;
        }

        return (
            <ToggleBox title="Related Articles" style="toggle-box--dark sp--dark-ui" isOpen={true}>
                {this.state.loading && <Loading />}
                <ul className="simple-list simple-list--dotted simple-list--no-padding">
                    {this.state.relatedArticles.map(article =>
                        <li className="simple-list__item" key={article.title}>
                            <p>{article.title}</p>
                            {this.props.rules.map(rule => {
                                let index = article.tenants.findIndex(t => t.code === rule.tenant.code);
                                let siteDomain = rule.tenant.subdomain ? rule.tenant.subdomain + '.' + rule.tenant.domain_name : rule.tenant.domain_name;

                                return <span key={'site' + rule.tenant.code} className={classNames('label-icon', { 'label-icon--success': index >= 0})}><i className="icon-globe"></i> {siteDomain}</span>
                            })}
                        </li>
                    )}
                </ul>
            </ToggleBox>
          );
    }
}
