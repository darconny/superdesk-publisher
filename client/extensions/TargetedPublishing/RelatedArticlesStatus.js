import React from 'react';
import axios from 'axios';
import classNames from 'classnames';
import {ToggleBox} from '../helperComponents/ToggleBox.jsx';
import Loading from '../helperComponents/Loading.jsx';

export default class RelatedArticlesStatus extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            item: props.item,
            apiUrl: props.apiUrl,
            apiHeader: props.apiHeader,
            rules: props.rules,
            relatedArticlesData: [],
            loading: true
        };
    }

    componentDidMount() {
        if (!this.state.item.associations) return;
        this.getRelatedArticlesData();
    }

    componentWillReceiveProps({rules}) {
        this.setState({rules});
    }

    getRelatedArticlesData = () => {
        return axios.post(this.state.apiUrl + 'organization/articles/related/', this.state.item, {headers: this.state.apiHeader})
            .then(res => {
                this.setState({
                    relatedArticlesData: res.data.related_article_items,
                    loading: false
                });
                return res;
            });
    }

    render() {

        if (!this.state.item.associations) return null;

        return (
            <ToggleBox title="Related Articles" style="toggle-box--dark sp--dark-ui" isOpen={true}>
                {this.state.loading && <Loading />}
                <ul className="simple-list simple-list--dotted simple-list--no-padding">
                    {this.state.relatedArticlesData.map(article =>
                        <li className="simple-list__item" key={article.title}>
                            <p>{article.title}</p>
                            {this.state.rules.map(rule => {
                                let index = article.tenants.findIndex(t => t.code === rule.tenant.code);
                                let siteDomain = rule.tenant.subdomain ? rule.tenant.subdomain + '.' + rule.tenant.domain_name : rule.tenant.domain_name;

                                return <span key={'site' + rule.tenant.code} className={classNames('label-icon', { 'label-icon--success': index >= 0})} className=""><i className="icon-globe"></i> {siteDomain}</span>

                            })}
                        </li>
                    )}
                </ul>
            </ToggleBox>
          );
    }
}
