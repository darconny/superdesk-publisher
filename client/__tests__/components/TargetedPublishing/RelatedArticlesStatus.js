import React from "react";
import RelatedArticlesStatus from "../../../components/TargetedPublishing/RelatedArticlesStatus";
import { render } from "@testing-library/react";
import axios from "axios";

jest.mock("axios");

describe("TargetedPublishing/RelatedArticlesStatus", () => {
  const item = {
    associations: true
  };

  const rules = [
    {
      tenant: {
        code: "eif0ca",
        subdomain: "tenant1",
        domain_name: "surcefabric.org"
      }
    }
  ];

  it("returns null when item has no associations", () => {
    const { container } = render(
      <RelatedArticlesStatus
        apiUrl="example.com/"
        apiHeader={{ Authrization: "Basic 1234567" }}
        rules={rules}
        item={{}}
      />
    );

    expect(container.firstChild).toBe(null);
  });

  it("renders correctly", async () => {
    const { container } = render(
      <RelatedArticlesStatus
        apiUrl="example.com/"
        apiHeader={{ Authrization: "Basic 1234567" }}
        rules={rules}
        item={item}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
