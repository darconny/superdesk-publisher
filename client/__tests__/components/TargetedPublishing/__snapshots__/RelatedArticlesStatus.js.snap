// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TargetedPublishing/RelatedArticlesStatus renders correctly 1`] = `
<div
  class="toggle-box toggle-box--dark sp--dark-ui"
>
  <div
    class="toggle-box__header"
  >
    <div
      class="toggle-box__chevron"
    >
      <i
        class="icon-chevron-right-thin"
      />
    </div>
    <div
      class="toggle-box__label"
    >
      Related Articles
    </div>
    <div
      class="toggle-box__line"
    />
  </div>
  <div
    class="toggle-box__content-wraper"
  >
    <div
      class="toggle-box__content"
    >
      <div
        class="spinner"
      >
        <div
          class="bounce1"
        />
        <div
          class="bounce2"
        />
        <div
          class="bounce3"
        />
      </div>
      <ul
        class="simple-list simple-list--dotted simple-list--no-padding"
      />
    </div>
  </div>
</div>
`;
